File: 10.php
<?php        

        $input = '1321131112';

        function process($string)
        {
            $i = 1;
            $total = strlen($string);
            $digits = str_split($string);
            $current = false;
            $count = false;
            $output = false;

            foreach ($digits as $d)
            {
                $last = $i === $total;

                if($current !== $d)
                {
                    if($current){ $output .= $count.$current; }

                    $current = $d;
                    $count = 1;
                }
                else
                {
                    $count++;
                }

                if($last){ $output .= $count.$current; }

                $i++;
            }

            return $output;
        }

        for($i=0; $i<50;$i++)
        {
            $input = process($input);
        }

        var_dump($input);

File: python/10.py
input = '1321131112'

def look_and_say(input):
	output = ''
	char = input[0]
	count = 1
	input = input[1:]
	cursor = 0

	for x in input:
		if char != x:
			output += str(count) + char
			char = x
			count = 1
			if cursor == len(input)-1: output += str(count) + char
		else:
			count += 1

		cursor += 1

	return output

for _ in range(40):
	input = look_and_say(input)

print(len(input))